static/app/views/profiling/profileDetails.tsx

```diff
@@ -149,6 +148,127 @@ function ProfileDetails() {
   );
 
+  const [filters, setFilters] = useState<Partial<Record<TableColumnKey, string[]>>>({});
+
+  const columnFilters = useMemo(() => {
+    function makeOnFilterChange(key: string) {
+      return values => {
+        setFilters(prevFilters => ({
+          ...prevFilters,
+          [key]: values.length > 0 ? values.map(val => val.value) : undefined,
+        }));
+      };
+    }
+    return {
+      type: {
+        values: ['application', 'system'],
+        onChange: makeOnFilterChange('type'),
+      },
+      image: {
+        values: pluckUniqueValues(slowestFunctions, 'image').sort((a, b) =>
+          a.localeCompare(b)
+        ),
+        onChange: makeOnFilterChange('image'),
+      },
+    };
+  }, [slowestFunctions]);
+
+  const currentSort = useMemo<GridColumnSortBy<TableColumnKey>>(() => {
+    let key = location.query?.functionsSort ?? '';
+
+    const defaultSort = {
+      key: 'self weight',
+      order: 'desc',
+    } as GridColumnSortBy<TableColumnKey>;
+
+    const isDesc = key[0] === '-';
+    if (isDesc) {
+      key = key.slice(1);
+    }
+
+    if (!key || !tableColumnKey.includes(key as TableColumnKey)) {
+      return defaultSort;
+    }
+
+    return {
+      key,
+      order: isDesc ? 'desc' : 'asc',
+    } as GridColumnSortBy<TableColumnKey>;
+  }, [location.query]);
+
+  useEffect(() => {
+    const removeListener = browserHistory.listenBefore((nextLocation, next) => {
+      if (location.pathname === nextLocation.pathname) {
+        next(nextLocation);
+        return;
+      }
+
+      if ('functionsSort' in nextLocation.query) {
+        delete nextLocation.query.functionsSort;
+      }
+
+      next(nextLocation);
+    });
+
+    return removeListener;
+  });
+
+  const generateSortLink = useCallback(
+    (column: TableColumnKey) => {
+      if (!SORTABLE_COLUMNS.has(column)) {
+        return () => undefined;
+      }
+      if (!currentSort) {
+        return () => ({
+          ...location,
+          query: {
+            ...location.query,
+            functionsSort: column,
+          },
+        });
+      }
+
+      const direction =
+        currentSort.key !== column
+          ? 'desc'
+          : currentSort.order === 'desc'
+          ? 'asc'
+          : 'desc';
+
+      return () => ({
+        ...location,
+        query: {
+          ...location.query,
+          functionsSort: `${direction === 'desc' ? '-' : ''}${column}`,
+        },
+      });
+    },
+    [location, currentSort]
+  );
+
+  const data = slowestFunctions
+    .filter(row => {
+      let include = true;
+      for (const key in filters) {
+        const values = filters[key];
+        if (!values) {
+          continue;
+        }
+        include = values.includes(row[key]);
+        if (!include) {
+          return false;
+        }
+      }
+      return include;
+    })
+    .sort((a, b) => {
+      if (currentSort.order === 'asc') {
+        return a[currentSort.key] - b[currentSort.key];
```

Comment:
Can we guarantee that a[key] or b[key] are not undefined here? This could otherwise result in some weird behavior :D 
<img width="227" alt="CleanShot 2022-10-17 at 09 20 44@2x" src="https://user-images.githubusercontent.com/9317857/196187776-da486ae2-80e3-45b3-bc86-c3f3bd90410f.png">